README.md

```diff
@@ -1,2 +1,214 @@
+		* In Runtime, used to make connections with Dawn and Shepherd.
+* **Shared Memory**
+	* These special files can be used for different processes to communicate messages, especially structured messages. Shared Memory is the fastest type of interprocess communication (IPC)
+	* In Runtime, it is used to enable the fast transfer of most data (especially the high-throughput data) through the system
+* **FIFO Pipes**
+    * These special files can be used for different processes to communicate messages, especially unstructured messages where the order in which the messages come is important
+	* In Runtime, used to transfer logs between the logger and `net_handler` before those logs are sent to Dawn, and used for `net_handler` and `executor` to communicate about the coding challenges
+* **Serial Communication**
+	* Serial communication happens whenever two devices communicate information byte-by-byte (usually over a wired connection), with no protocol or breaking up of large messages into smaller packets when necessary
+	* In Runtime, `dev_handler` and the `lowcar` devices communicate over serial connections
+* **Thread Management**
+	* Threads are essentially ways to run different sections of code within the same process simultaneously. They are very useful for breaking up the work a process needs to do into several different, well-defined "parts"
+	* In Runtime, they are used by all processes to divide up their tasks appropriately and prevent logic in one section of code from crashing or blocking logic in another section
+* **Process Management**
+	* Processes can also spawn and kill other processes. Doing this is useful when you want to enforce a separation between two processes, and creating a new thread is not enough of a separation; or when you want to run a separate piece of code and ensure that everything related to that process is stopped cleanly when it exits
+	* In Runtime, they are used by `executor` to run the student code, to ensure that nothing that is done in student code can affect the `dev_handler` and make it crash
+
+## Dependencies
+
+Remember, one of the design principles is KISS. Nevertheless, Runtime is a complex system and will require effort to learn and understand. Hopefully this list does not seem too daunting to you! <sup id="return3">[3](#footnote3)</sup>
+
+As a baseline, Runtime uses the following commonly used tools that should already be installed on your machine (if you don't have these, ask the Runtime Project Manager to help you install them):
+
+* Python 3.7 or later: students write their student code in Python, so in order to run their code, we do need to use some Python in `executor`. To run Python code from C, we need to use Python's C API. Python's C API is notoriously unstable, and we wrote `executor` using Python 3.7 / 3.8.
```

Comment:
I wouldn't say the API is unstable at all, probably the opposite. Its just always good to specify versions though for the future. I would just say 3.7 for now, though I think 3.6 and 3.8 would both work.

Reply:
👌